pkg/instrumentation/auto/util_test.go

```diff
@@ -0,0 +1,154 @@
+			autoMonitorConfig:    `{"monitorAllServices":true}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   false,
+			expectedType:         "*auto.AnnotationMutators",
+		},
+		{
+			name:                 "Annotation disabled, Monitor enabled with monitorAllServices=true",
+			envDisableAnnotation: true,
+			envDisableMonitor:    false,
+			autoAnnotationConfig: `{"java":{"deployments":["default/myapp"]}}`,
+			autoMonitorConfig:    `{"monitorAllServices":true}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   true,
+			expectedType:         "*auto.Monitor",
+		},
+		{
+			name:                 "Annotation disabled, Monitor enabled with monitorAllServices=false",
+			envDisableAnnotation: true,
+			envDisableMonitor:    false,
+			autoAnnotationConfig: `{"java":{"deployments":["default/myapp"]}}`,
+			autoMonitorConfig:    `{"monitorAllServices":false}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   false,
+			expectedType:         "*auto.Monitor",
+		},
+		{
+			name:                 "Invalid annotation config, valid monitor config",
+			envDisableAnnotation: false,
+			envDisableMonitor:    false,
+			autoAnnotationConfig: `{invalid-json}`,
+			autoMonitorConfig:    `{"monitorAllServices":true}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   true,
+			expectedType:         "*auto.Monitor",
+		},
+		{
+			name:                 "Empty annotation config, valid monitor config",
+			envDisableAnnotation: false,
+			envDisableMonitor:    false,
+			autoAnnotationConfig: `{}`,
+			autoMonitorConfig:    `{"monitorAllServices":true}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   true,
+			expectedType:         "*auto.Monitor",
+		},
+		{
+			name:                 "Valid annotation config, invalid monitor config",
+			envDisableAnnotation: false,
+			envDisableMonitor:    false,
+			autoAnnotationConfig: `{"java":{"deployments":["default/myapp"]}}`,
+			autoMonitorConfig:    `{invalid-json}`,
+			expectNilAnnotator:   false,
+			expectedMonitorAll:   false,
+			expectedType:         "*auto.AnnotationMutators",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Set environment variables
+			if tt.envDisableAnnotation {
+				os.Setenv("DISABLE_AUTO_ANNOTATION", "true")
+			} else {
+				os.Unsetenv("DISABLE_AUTO_ANNOTATION")
+			}
+
+			if tt.envDisableMonitor {
+				os.Setenv("DISABLE_AUTO_MONITOR", "true")
+			} else {
+				os.Unsetenv("DISABLE_AUTO_MONITOR")
+			}
+
+			// Call the function
+			annotator, monitorAll := createInstrumentationAnnotatorWithClientset(tt.autoMonitorConfig, tt.autoAnnotationConfig, ctx, fake.NewSimpleClientset(), fakeClient, fakeClient, logger)
+
+			// Check results
+			if tt.expectNilAnnotator {
+				assert.Nil(t, annotator, "Expected nil annotator")
+			} else {
+				assert.NotNil(t, annotator, "Expected non-nil annotator")
+
+				// Check type using type assertion
+				actualType := fmt.Sprintf("%T", annotator)
```

Comment:
nit: Would prefer to use reflection or some enum that checks the actual type instead of these string representations. If we ever rename the struct or package or move it to a different package, these tests will break.

Could do something like
```
expectedType: reflect.TypeOf(auto.AnnotationMutators{})

...

assert.Equals(t, tt.expectedType, reflect.TypeOf(annotator))
```

or

```go
type annotatorType int

const (
	unset annotatorType = iota
	mutators
	monitor
)

...

switch annotator.(type) {
case *auto.AnnotationMutators:
  assert.Equals(t, mutators, tt.expectedType)
}
```